### ISSUE TITLE
Investigate ExportAnnotations a11y test failing when using the new SelectNext with role="combobox"

### ISSUE THREAD

**Issue by acelaya** [hypothesis/client#5939]
> Depends on https://github.com/hypothesis/client/pull/5938

Recently, we changed `SelectNext` in frontend-shared so that the toggle button had `role="combobox"`.

This solved how the component was announced by screen readers when used in combination with label elements, but apparently, axe-core considers that invalid, and requires elements with `role="combobox"` to be labelled via `aria-label` or `aria-labelledby`.

Let's investigate if this is actually correct, or an issue in axe-core.

**Comment by acelaya** [comment]
After some further investigation, I tend to think our approach should be considered valid.

In [this document](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) there's a paragraph about how comboboxes should be labelled, that reads:

> If the combobox has a visible label and the combobox element is an HTML element that can be labelled using the HTML label element (e.g., the input element), it is labeled using the label element. Otherwise, if it has a visible label, the combobox element has [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the labelling element. Otherwise, the combobox element has a label provided by [aria-label](https://w3c.github.io/aria/#aria-label).

The first part is the one that applies to us. The `SelectNext` component wraps a `<button />` with an optional `id` attribute, that can then be referenced via `for` attribute from a label.

According to this [other document](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label), buttons are among the valid elements to be referenced by labels:

> Elements that can be associated with a `<label>` element include `<button>`, `<input>` (except for type="hidden"), `<meter>`, `<output>`, `<progress>`, `<select>` and `<textarea>`.

**Comment by acelaya** [comment]
I have reported this to axe-core just reported it https://github.com/dequelabs/axe-core/issues/4235
That provides all the information needed to reproduce this.
